lib/hanami/events/matcher.rb

```diff
@@ -0,0 +1,24 @@
+module Hanami
+  module Events
+    class Matcher
+      MATCH_ALL_CHARS = '.*'
+      RANGE_PATTERN = '*'
+
+      def initialize(pattern)
+        @pattern = Regexp.new(to_regexp_string(pattern))
+      end
+
+      def match?(event_name)
+        !!@pattern.match(event_name)
+      end
+
+      private
+
+      def to_regexp_string(pattern)
+        pattern.split('.')
+          .map { |p| p == RANGE_PATTERN ? MATCH_ALL_CHARS : p }
```

Comment:
I would escape anything that could potentially break regex using `Regexp.escape(p)`:
```ruby
pattern.split('.')
  .map { |p| p == RANGE_PATTERN ? MATCH_ALL_CHARS : Regexp.escape(p) }.join('\.')
```

Without escaping, this event `user.te*st.created` wouldn't be matched by the same pattern `user.te*st.created`. Kind of edge case, but it's better to be prepared.